src/clients/resp.rs

```diff
@@ -0,0 +1,997 @@
+use super::*;
+use crate::net::Connector;
+use crate::Instant;
+use redis::{AsyncCommands, RedisConnectionInfo};
+use std::borrow::Borrow;
+
+/// Launch tasks with one conncetion per task as RESP protocol is not mux-enabled.
+pub fn launch_tasks(runtime: &mut Runtime, config: Config, work_receiver: Receiver<WorkItem>) {
+    debug!("launching resp protocol tasks");
+
+    // create one task per "connection"
+    // note: these may be channels instead of connections for multiplexed protocols
+    for _ in 0..config.client().unwrap().poolsize() {
+        for endpoint in config.target().endpoints() {
+            runtime.spawn(task(
+                work_receiver.clone(),
+                endpoint.clone(),
+                config.clone(),
+            ));
+        }
+    }
+}
+
+#[allow(dead_code)]
+#[allow(clippy::slow_vector_initialization)]
+async fn task(work_receiver: Receiver<WorkItem>, endpoint: String, config: Config) -> Result<()> {
```

Comment:
I think this function is too long. The commands are better grouped by data type and put in their own functions, or even their own files (one per data structure, for example).

Reply:
Maybe we need a command family enum to reduce the length of the match. Would make it cleaner to split and group by family.